openlibrary/core/lending.py

```diff
@@ -211,7 +216,13 @@ def get_available(limit=None, page=1, subject=None, query=None,
     """
     url = url or compose_ia_url(
         limit=limit, page=page, subject=subject, query=query,
-        work_id=work_id, _type=_type, sorts=sorts)
+        work_id=work_id, _type=_type, sorts=sorts
+    )
+    assert url, (
```

Comment:
This assertion seems to be sporadically failing on dev `/opt/openlibrary/deploys/openlibrary/openlibrary/openlibrary/templates/home/custom_ia_carousel.html: error in processing template: AssertionError: get_available(limit=42, page=1, subject=, query=, work_id=/works/OL14853604W, _type=subjects, sorts=, url=) (falling back to default template)`

Reply:
Op, or consistently: https://dev.openlibrary.org/books/OL8687251M/The_Ku_Klux_Klan?debug=true

Reply:
So for this [KKK book](https://dev.openlibrary.org/books/OL8687251M.json) and its [author](https://dev.openlibrary.org//authors/OL20753A.json)...  Why are they not being found?  Would it be better to raise the assertion on line 171 or on 221?  Or should we even raise an assertion on line 153 if `not works_authors_and_subjects`?  My sense is the earlier in the execution flow the better.

Reply:
Oy, I just picked that book at random _from the homepage_ 😬.

We shouldn't be `assert`-ing; just logging the exception. I can get a grep of matching log outputs from prod, which we can then analyze to try to figure out why things are failing and how to fix them. Does that seem good?